jetty-core/jetty-io/src/main/java/org/eclipse/jetty/io/ArrayByteBufferPool.java

```diff
@@ -415,14 +330,13 @@ public long getHeapMemory()
 
     private long getMemory(boolean direct)
     {
-        AtomicLong memory = direct ? _directMemory : _heapMemory;
-        return memory.longValue();
-    }
-
-    private void subtractMemory(int amount, boolean direct)
-    {
-        AtomicLong memory = direct ? _directMemory : _heapMemory;
-        memory.addAndGet(-amount);
+        RetainedBucket[] buckets = direct ? _direct : _indirect;
+        long total = 0L;
+        for (RetainedBucket bucket : buckets)
+        {
+            total += bucket._pool.size() * (long)bucket._capacity;
```

Comment:
I'm not sure what this number is?    It is all idle and inuse buffers from any concurrent pool plus just the idle buffers from any queued pool.